gcp_variant_transforms/testing/temp_dir.py

```diff
@@ -40,20 +44,38 @@ def get_path(self):
     """Returns the path to the temporary directory."""
     return self._tempdir
 
-  def create_temp_file(self, suffix='', lines=None):
+  def create_temp_file(
+      self, suffix='', lines=None,
+      compression_type=filesystem.CompressionTypes.UNCOMPRESSED):
     """Creates a temporary file in the temporary directory.
+
     Args:
       suffix (str): The filename suffix of the temporary file (e.g. '.txt')
       lines (List[str]): A list of lines that will be written to the temporary
         file.
+      compression_type (``CompressionTypes``): Enum specifying compression type.
     Returns:
       The name of the temporary file created.
+    Raises:
+      ValueError: If ``compression_type`` is unsupported.
     """
     f = tempfile.NamedTemporaryFile(delete=False,
                                     dir=self._tempdir,
                                     suffix=suffix)
-    if lines:
-      for line in lines:
-        f.write(line)
+    if not lines:
+      return f.name
+    if compression_type in (filesystem.CompressionTypes.UNCOMPRESSED,
+                            filesystem.CompressionTypes.AUTO):
+      f.write(''.join(lines))
+    elif compression_type == filesystem.CompressionTypes.GZIP:
+      f.close()
```

Comment:
temp files are a bit weird and can actually be read and written without being closed, so I think this line should be removed. In fact, the API specifies that the file will be destroyed upon being closed, so the current implementation is probably destroying and recreating a normal file with the same name.

Reply:
ah! nice catch! removed.